src/SFA.DAS.LevyTransferMatching.Web/Orchestrators/OpportunitiesOrchestrator.cs

```diff
@@ -134,12 +160,73 @@ public async Task<ApplyViewModel> GetApplyViewModel(ApplicationRequest request)
                 Sectors = "-",
                 Locations = "-",
                 MoreDetail = "-",
-                ContactName = "-",
-                EmailAddress = "-",
-                WebsiteUrl = "-"
+                ContactName = string.IsNullOrWhiteSpace(contactName) ? "-" : contactName,
+                EmailAddresses = emailAddresses,
+                WebsiteUrl = string.IsNullOrEmpty(application.BusinessWebsite) ? "-" : application.BusinessWebsite,
             };
         }
 
+        public async Task<ContactDetailsViewModel> GetContactDetailsViewModel(ContactDetailsRequest contactDetailsRequest)
+        {
+            var opportunityDto = await _opportunitiesService.GetOpportunity(contactDetailsRequest.PledgeId);
+
+            if (opportunityDto == null)
+            {
+                return null;
+            }
+
+            var encodedPledgeId = _encodingService.Encode(opportunityDto.Id, EncodingType.PledgeId);
+
+            var opportunitySummaryViewModel = await GetOpportunitySummaryViewModel(opportunityDto, encodedPledgeId);
+
+            var cacheItem = await this.RetrieveCacheItem(contactDetailsRequest.CacheKey);
+
+            IEnumerable<string> additionalEmailAddresses = cacheItem.AdditionalEmailAddresses;
```

Comment:
You could probably tuck some of this code away into a method on the cache item, to make it read more cleanly.

Reply:
I've moved this code into the `get`ter for `AdditionalEmailAddresses` on the `cacheItem`.